### PR TITLE
Exclude small-channel Conv nodes from FP8 quantization

### PR DESCRIPTION
**Pull request by nv-samcheng**
### What does this PR do?

Small-channel convolutions (IC or OC <= 16) do not benefit from FP8 quantization and cause performance regressions due to poor kernel selection. Hardcode the threshold directly in graph_utils rather than exposing it as a CLI flag.

### Usage

```python
# Add a code snippet demonstrating how to use this
```

### Before your PR is "*Ready for review*"

Make sure you read and follow [Contributor guidelines](https://github.com/NVIDIA/Model-Optimizer/blob/main/CONTRIBUTING.md) and your commits are signed (`git commit -s -S`).

Make sure you read and follow the [Security Best Practices](https://github.com/NVIDIA/Model-Optimizer/blob/main/SECURITY.md#security-coding-practices-for-contributors) (e.g. avoiding hardcoded `trust_remote_code=True`, `torch.load(..., weights_only=False)`, `pickle`, etc.).

- Is this change backward compatible?: ✅ / ❌ / N/A 
- If you copied code from any other sources or added a new PIP dependency, did you follow guidance in `CONTRIBUTING.md`: ✅ / ❌ / N/A 
- Did you write any new necessary tests?: ✅ / ❌ / N/A 
- Did you update [Changelog](https://github.com/NVIDIA/Model-Optimizer/blob/main/CHANGELOG.rst)?: ✅ / ❌ / N/A 

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved FP8 quantization filtering to exclude convolutions with very small input or output channel counts, reducing incorrect FP8 quantization and improving model accuracy and stability.

* **Tests**
  * Added unit tests validating FP8 small-channel exclusion across channel combinations and confirming no change in int8 mode.

### CODE REVIEW

File: modelopt/onnx/quantization/graph_utils.py [hunk 1]
```diff
@@ -1147,6 +1152,19 @@ def find_nodes_from_convs_to_exclude(graph: Graph, quantize_mode: str = "int8"):
             if quantize_mode == "fp8" and filter_size > 32:
                 logger.debug(f"Found large filter conv for FP8: {node.name}")
                 unsupported_conv_nodes.append(node.name)
+                continue
```
Comment (ajrasane) [inline modelopt/onnx/quantization/graph_utils.py:1216]: Could you add a comment explaining why we continue here now?

File: modelopt/onnx/quantization/graph_utils.py
Comment (ajrasane) [inline modelopt/onnx/quantization/graph_utils.py:1184]: Isn't the change below already handled here for FP8? Should we change this to <= instead?
If not, could you add a test for an edge case between these two paths?

File: tests/unit/onnx/quantization/test_graph_utils.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (ajrasane) [inline tests/unit/onnx/quantization/test_graph_utils.py]: nit: I think the license year should be 2026 now right?

File: tests/unit/onnx/quantization/test_graph_utils.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    excluded = find_nodes_from_convs_to_exclude(graph, quantize_mode="fp8")
+    if expected_excluded:
+        assert "Conv_0" in excluded
+    else:
+        assert "Conv_0" not in excluded
+
+
+def test_fp8_small_channel_exclusion_does_not_affect_int8():
+    """The small-channel FP8 exclusion should not apply in int8 mode."""
+    graph = _make_conv_graph(output_channels=16, input_channels=64, kernel_shape=(3, 3))
```
Comment (ajrasane) [inline tests/unit/onnx/quantization/test_graph_utils.py]: Should we set this as 8,64 instead so that it passes the int8 check?

File: modelopt/onnx/quantization/graph_utils.py (hunk 1 above)
Comment (nv-samcheng) [reply modelopt/onnx/quantization/graph_utils.py:1216]: Done

File: modelopt/onnx/quantization/graph_utils.py
Comment (nv-samcheng) [reply modelopt/onnx/quantization/graph_utils.py:1184]: Yes, this is for int8 case, we should seperate them for different cases.
I'll add edge case to cover it.

File: tests/unit/onnx/quantization/test_graph_utils.py (hunk 3 above)
Comment (nv-samcheng) [reply tests/unit/onnx/quantization/test_graph_utils.py]: Yes. updated. Thanks.

File: tests/unit/onnx/quantization/test_graph_utils.py (hunk 2 above)
Comment (nv-samcheng) [reply tests/unit/onnx/quantization/test_graph_utils.py]: Nice catch, just copy from other file. thanks.

**Comment by nv-samcheng** [comment]
Fixed, @ajrasane thanks for reviewing, please let me know if there is other suggestion or questions

**Comment by nv-samcheng** [comment]
the CI broken. https://github.com/NVIDIA/Model-Optimizer/actions/runs/23554272151/job/68581326529?pr=1083
@ajrasane Do you know if this related with my changing? or infra issues?
